internal/store/bundle_test.go

```diff
@@ -290,3 +292,179 @@ func TestRemove(t *testing.T) {
 		assert.Equal(t, len(bundles), 0)
 	})
 }
+
+func TestLookUp(t *testing.T) {
+	dockerConfigDir := fs.NewDir(t, t.Name(), fs.WithMode(0755))
+	defer dockerConfigDir.Remove()
+	appstore, err := NewApplicationStore(dockerConfigDir.Path())
+	assert.NilError(t, err)
+	bundleStore, err := appstore.BundleStore()
+	assert.NilError(t, err)
+	bndl := &bundle.Bundle{Name: "bundle-name"}
+	id, err := bundleStore.Store(nil, bndl)
+	assert.NilError(t, err)
+	ref := parseRefOrDie(t, "my-repo/a-bundle:my-tag")
+	_, err = bundleStore.Store(ref, bndl)
+	assert.NilError(t, err)
+	dockerIoRef := parseRefOrDie(t, "docker.io/library/a-bundle:my-tag")
+	_, err = bundleStore.Store(dockerIoRef, bndl)
+	assert.NilError(t, err)
+
+	for _, tc := range []struct {
+		Name          string
+		refOrID       string
+		ExpectedError string
+		ExpectedRef   reference.Reference
+	}{
+		{
+			Name:          "Long Id",
+			refOrID:       id.String(),
+			ExpectedError: "",
+			ExpectedRef:   id,
+		},
+		{
+			Name:          "Short Id",
+			refOrID:       id.String()[0:8],
+			ExpectedError: "",
+			ExpectedRef:   id,
+		},
+		{
+			Name:          "Tagged Ref",
+			refOrID:       "my-repo/a-bundle:my-tag",
+			ExpectedError: "",
+			ExpectedRef:   ref,
+		},
+		{
+			Name:          "docker.io_library repository Tagged Ref",
+			refOrID:       "a-bundle:my-tag",
+			ExpectedError: "",
+			ExpectedRef:   dockerIoRef,
+		},
+		{
+			Name:          "Unknown Tag",
+			refOrID:       "other-repo/a-bundle:other-tag",
+			ExpectedError: "docker.io/other-repo/a-bundle:other-tag: reference not found",
+			ExpectedRef:   nil,
+		},
+		{
+			Name:          "Unknown ID",
+			refOrID:       "b4fcc3af16804e29d977918a3a322daf1eb6ab2992c3cc7cbfeae8c3d6ede8af",
+			ExpectedError: "b4fcc3af16804e29d977918a3a322daf1eb6ab2992c3cc7cbfeae8c3d6ede8af: reference not found",
+			ExpectedRef:   nil,
+		},
+		{
+			Name:          "Unknown short ID",
+			refOrID:       "b4fcc3af",
+			ExpectedError: "docker.io/library/b4fcc3af:latest: reference not found",
+			ExpectedRef:   nil,
+		},
+	} {
+		t.Run(tc.Name, func(t *testing.T) {
+			fmt.Println(tc.refOrID)
+			ref, err := bundleStore.LookUp(tc.refOrID)
+
+			if tc.ExpectedError != "" {
+				assert.Equal(t, err.Error(), tc.ExpectedError)
+			} else {
+				assert.NilError(t, err)
+			}
+
+			if tc.ExpectedRef != nil {
+				assert.Equal(t, ref, tc.ExpectedRef)
+			}
+		})
+	}
+}
+
+func TestScanBundles(t *testing.T) {
+	dockerConfigDir := fs.NewDir(t, t.Name(), fs.WithMode(0755))
+	//defer dockerConfigDir.Remove()
+
+	bndl1 := &bundle.Bundle{Name: "bundle-1"}
```

Comment:
You should add some comment per block, explaining what you are doing here
// Preparing a simple bundle stored by id
// Preparing a simple bundle stored by id and ref
etc...